consensus/propagation/types/types.go

```diff
@@ -66,6 +70,40 @@ type CompactBlock struct {
 	proofsCache []*merkle.Proof
 }
 
+// SignBytes returns the compact block commitment data that
+// needs to be signed.
+// The sign bytes are the sha256 hash over the following
+// concatenated data:
+// - BpHash
+// - Blobs
+// - Proposal.Signature
+// - Big endian encoding of LastLen
+// - PartsHashes
+func (c *CompactBlock) SignBytes() ([]byte, error) {
+	bytes := make([]byte, 0)
+	bytes = append(bytes, c.BpHash...)
+	for _, md := range c.Blobs {
+		pb, err := proto.Marshal(md.ToProto())
+		if err != nil {
+			return nil, err
+		}
+		bytes = append(bytes, pb...)
+	}
+	bytes = append(bytes, c.Proposal.Signature...)
+
+	// big endian encode the last len
+	lastLenBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(lastLenBytes, c.LastLen)
+	bytes = append(bytes, lastLenBytes...)
+
+	for _, ph := range c.PartsHashes {
+		bytes = append(bytes, ph...)
+	}
+
+	signBytes := tmhash.Sum(bytes)
+	return signBytes, nil
+}
```

Comment:
just checking the logic here, we're including the signature of the proposal and not important things from the proposal cause they're already signed over?

Reply:
yes